ImperatorToCK3/Source/Mappers/RegionMapper/CK3Region.h

```diff
@@ -0,0 +1,35 @@
+#ifndef CK3_REGION_H
+#define CK3_REGION_H
+
+#include "Parser.h"
+#include <set>
+#include "../../CK3/Titles/Title.h"
+#include "Log.h"
+
+namespace mappers
+{
+class CK3Region: commonItems::parser
+{
+  public:
+	explicit CK3Region(std::istream& theStream);
+
+	[[nodiscard]] const auto& getRegions() const { return regions; }
+	[[nodiscard]] const auto& getDuchies() const { return duchies; }
+	[[nodiscard]] const auto& getCounties() const { return counties; }
+	[[nodiscard]] const auto& getProvinces() const { return provinces; }
+	[[nodiscard]] bool regionContainsProvince(unsigned long long province) const;
+
+	void linkRegion(const std::pair<std::string, std::shared_ptr<CK3Region>>& theRegion) { regions[theRegion.first] = theRegion.second; }
```

Comment:
Take two parameters instead of a pair. Not only is it inefficient to pack and immediately unpack, it obscures the intent by removing variable names.

Reply:
We had this discussion before where I refused to do so arguing the pairs are what converter (ck3toeu4) operate on. I shove pairs left or right knowing first is the key and second is the pointer. I still stand by that, but in retrospect (after you commented recently on naming a small struct and moving *that* instead of a nameless pair is better, I tend to agree. Of course that doesn't apply here as it'd require a lot of rework, but if I do a revision of ck3 code I'll probably replace all those pairs with ministructs that serve the same function.